core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -178,6 +180,16 @@ public static String getSchemaFilePath(String tablePath) {
     return getActualSchemaFilePath(tablePath);
   }
 
+  /**
+   * return the schema file path
+   * @param tablePath path to table files
+   * @param hadoopConf hadoop configuration object
+   * @return schema file path
+   */
+  public static String getSchemaFilePath(String tablePath, Configuration hadoopConf) {
```

Comment:
why add two function just for a single line, can keep a single function to avoid confusion

Reply:
If you check the caller old getSchemaFilePath, many places it is used. cannot pass null config in all the places. 
That's why used method overloading concept as it is applicable here

Reply:
one private and one public method it is. used a same private method to handle both the public method signatures